cmd/goal/account.go

```diff
@@ -877,11 +877,16 @@ var addParticipationKeyCmd = &cobra.Command{
 		// Generate a participation keys database and install it
 		client := ensureFullClient(dataDir)
 
-		_, _, err := client.GenParticipationKeysTo(accountAddress, roundFirstValid, roundLastValid, keyDilution, partKeyOutDir)
-		if err != nil {
-			reportErrorf(errorRequestFail, err)
+		participationGen := func() {
+			_, _, err := client.GenParticipationKeysTo(accountAddress, roundFirstValid, roundLastValid, keyDilution, partKeyOutDir)
+			if err != nil {
+				reportErrorf(errorRequestFail, err)
+			}
+			reportInfof("Participation key generation successful")
```

Comment:
same as above; please place the error reporting after the `util.RunFuncWithSpinningCursor`